### PR TITLE
Add concept introduction templates

### LINKED ISSUE #2077: Add introduction.md.tpl files for concept exercises
Similar to exercism/fsharp#1145

### PR DESCRIPTION
**Pull request by glennj**
The order of the included concept introductions has changed for some of the exercises. If it's problematic, I can manually alter the template files.

Closes #2077

### CODE REVIEW

**Comment by glennj** [comment]
I see that some of the .tpl files did not generate .md files. I can't see anything in config.json that would preclude that (wip status for instance)

File: exercises/concept/bird-watcher/.docs/introduction.md.tpl [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+%{concept:arrays}
+
+## For Loops
+
+%{concept:for-loops}
+
+## Foreach Loops
+
+%{concept:foreach-loops}
```
Comment (ErikSchierboom) [inline exercises/concept/bird-watcher/.docs/introduction.md.tpl:13]: This new ordering makes more sense I feel!

File: exercises/concept/building-telemetry/.docs/introduction.md [hunk 2]
```diff
@@ -1,10 +1,12 @@
 # Introduction
 
+## Parameters
+
 This exercise discusses some details of method parameters and their use in C#.
 
 Parameters convey information from a calling method to a called method.
 
-- The behavior of parameters can be modified by the use of modifies such as `ref` and `out`.
+- The behavior of parameters can be modified by the use of modifiers such as `ref` and `out`.
```
Comment (ErikSchierboom) [inline exercises/concept/building-telemetry/.docs/introduction.md:9]: Thanks!

File: exercises/concept/cars-assemble/.docs/introduction.md.tpl [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Introduction
+
+## If Statements
+
+%{concept:if-statements}
+
+## Numbers
+
+%{concept:numbers}
```
Comment (ErikSchierboom) [inline exercises/concept/cars-assemble/.docs/introduction.md.tpl:9]: Maybe swap these?

```suggestion
## Numbers

%{concept:numbers}

## If Statements

%{concept:if-statements}
```

File: exercises/concept/developer-privileges/.docs/introduction.md [hunk 4]
```diff
@@ -18,7 +20,7 @@ Collections can also be initialized in this way. Typically, this is accomplished
 IList<Person> people = new List<Person>{ new Person(), new Person{Name="Joe Shmo"}};
 ```
 
-Dictionaries use the following syntax:
+Dictinaries use the following syntax:
```
Comment (ErikSchierboom) [inline exercises/concept/developer-privileges/.docs/introduction.md:23]:
```suggestion
Dictionaries use the following syntax:
```

File: exercises/concept/faceid-2/.docs/introduction.md [hunk 5]
```diff
@@ -24,4 +24,4 @@ The values used in the equality test must be stable while the hashed collection
 
 ## Sets
 
-The `HashSet` library class provides a good mechanism for storing unique values.
+A set is a collection of unique values. The default .NET implementation of a set is the `HashSet<T>` class, which is a collection that only contains unique values.
```
Comment (ErikSchierboom) [inline exercises/concept/faceid-2/.docs/introduction.md:27]: Great!

File: exercises/concept/interest-is-interesting/.docs/introduction.md.tpl [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+%{concept:do-while-loops}
+
+## Floating Point Numbers
+
+%{concept:floating-point-numbers}
+
+## While Loops
+
+%{concept:while-loops}
```
Comment (ErikSchierboom) [inline exercises/concept/interest-is-interesting/.docs/introduction.md.tpl:13]: Maybe reorder to:

```suggestion
## Floating Point Numbers

%{concept:floating-point-numbers}

## While Loops

%{concept:while-loops}

## Do While Loops

%{concept:do-while-loops}
```

File: exercises/concept/need-for-speed/.docs/introduction.md [hunk 7]
```diff
@@ -1,11 +1,13 @@
...
 
+## Constructors
+
 Creating an instance of a _class_ is done by calling its _constructor_ through the `new` operator. A constructor is a special type of method whose goal is to initialize a newly created instance. Constructors look like regular methods, but without a return type and with a name that matches the classes' name.
 
 ```csharp
 class Library
 {
-    private books;
+    private int books;
```
Comment (ErikSchierboom) [inline exercises/concept/need-for-speed/.docs/introduction.md:10]: Doh!

File: exercises/concept/remote-control-competition/.docs/introduction.md [hunk 8]
```diff
@@ -37,4 +37,11 @@ Interfaces can contain instance methods and properties amongst other members
...
 
-The `IComparable<T>` interface can be implemented where a default generic sort order in collections is required.
+To allow an object to be compared to another object, it must implement the `IComparable<T>` interface.
+This interface has a single method that needs to be implemented: `int CompareTo(T other)`.
+
+There are three possible return values for the `CompareTo` method:
+
+- Value smaller than zero: the current object is smaller than the other object.
+- Value greater than zero: the current object is greater than the other object.
+- Value is zero: the current object is equal to the other object.
```
Comment (ErikSchierboom) [inline exercises/concept/remote-control-competition/.docs/introduction.md:47]: Excellent!

File: exercises/concept/weighing-machine/.docs/introduction.md [hunk 9]
```diff
@@ -20,18 +22,18 @@ accessor and vice versa. A property doesn't have to have both accessors, it can
...
 
-## Field/Expression Backed Properties:
+### Field/Expression Backed Properties:
 
 ```csharp
 private int myField;
 public int MyProperty
 {
-    get { return myField; }
+    get { return myfField; }
```
Comment (ErikSchierboom) [inline exercises/concept/weighing-machine/.docs/introduction.md:31]:
```suggestion
    get { return myField; }
```

**Review by ErikSchierboom — COMMENTED** [review]
> I see that some of the .tpl files did not generate .md files -- attack-of-the-trolls for example. I can't see anything in config.json that would preclude that.

I think they were generated but they didn't change.

**Comment by ErikSchierboom** [comment]
Thanks for this huge undertaking! Much appreciated. Just a couple of minor nits.

**Comment by glennj** [comment]
Thanks for the very careful review.

File: exercises/concept/weighing-machine/.docs/introduction.md [hunk 10]
```diff
@@ -1,9 +1,11 @@
 # Introduction
 
+## Properties
+
 A property in C# is a member of a class that provides access to data within that class.
 Callers can set or retrieve (get) the data. Properties can be either auto-implemented or
 have a backing field. They comprise a set accessor and/or a get accessor.
-In some other languages a "mutator" is roughly equivalent to 
+In some other languages a "mutator" is roughly equivalent to a
```
Comment (ErikSchierboom) [inline exercises/concept/weighing-machine/.docs/introduction.md:8]:
```suggestion
In some other languages a "mutator" is roughly equivalent to
```

**Review by ErikSchierboom — COMMENTED** [review]
One tiny nit. Sorry!

**Review by ErikSchierboom — APPROVED** [review]
Awesome! Thanks a lot

**Pull request merged**